src/AdminConsole/Components/Pages/Organization/Admins.razor.cs

```diff
@@ -0,0 +1,150 @@
+
+    private readonly IReadOnlyCollection<string> _activeColumnHeaders = new List<string> { "Admin", "E-mail", "Action" };
+    private readonly IReadOnlyCollection<string> _invitationColumnHeaders = new List<string> { "Recipient", "Invited", "Status", "Sender", "Action" };
+
+    private const string DeleteActiveFormName = "delete-active-form";
+    private const string InviteFormName = "invite-form";
+    private const string CancelInviteFormName = "cancel-invite-form";
+
+    private ModelState? DeleteAdminModelState { get; set; }
+    private ModelState? InviteAdminModelState { get; set; }
+
+    [SupplyParameterFromForm(FormName = DeleteActiveFormName)]
+    public DeleteActiveFormModel DeleteActiveForm { get; set; } = new();
+
+    [SupplyParameterFromForm(FormName = InviteFormName)]
+    public InviteFormModel InviteForm { get; set; } = new();
+
+    [SupplyParameterFromForm(FormName = CancelInviteFormName)]
+    public CancelInviteFormModel CancelInviteForm { get; set; } = new();
+
+    public bool CanInvite { get; set; }
+
+    public IReadOnlyCollection<ConsoleAdmin>? ConsoleAdmins { get; set; }
+
+    public IReadOnlyCollection<Invite>? Invites { get; set; }
+
+    protected override async Task OnInitializedAsync()
+    {
+        ConsoleAdmins = await DataService.GetConsoleAdminsAsync();
+        Invites = await InvitationService.GetInvitesAsync(CurrentContext.OrgId!.Value);
+        CanInvite = await DataService.CanInviteAdminAsync();
+    }
+
+    private async Task DeleteAdminAsync()
+    {
+        var users = await DataService.GetConsoleAdminsAsync();
+        if (users is not { Count: > 1 })
+        {
+            DeleteAdminModelState = new ModelState("At least one admin is required in an organization.");
+            return;
+        }
+
+        var user = users.FirstOrDefault(u => u.Id == DeleteActiveForm.UserId);
+        if (user is null)
+        {
+            DeleteAdminModelState = new ModelState("User not found.");
+            return;
+        }
+
+        await PasswordlessClient.DeleteUserAsync(user.Id);
+        await UserManager.DeleteAsync(user);
+
+        var performedBy = users.First(x => x.Email == HttpContextAccessor.HttpContext!.User.GetEmail());
+        EventLogger.LogDeleteAdminEvent(performedBy, user, TimeProvider.GetUtcNow().UtcDateTime);
+
+        // if user is self
+        if (user.Email == HttpContextAccessor.HttpContext!.User.GetEmail())
+        {
+            await SigninManager.SignOutAsync();
+            NavigationManager.NavigateTo("/Account/Login");
+        }
+
+        NavigationManager.Refresh();
+    }
+
+    private async Task InviteAsync()
+    {
+        if (CanInvite is false)
+        {
+            InviteAdminModelState = new ModelState("You need to upgrade to a paid organization to invite more admins.");
+            return;
+        }
+
+        if (ConsoleAdmins!.Any(x => string.Equals(x.Email, InviteForm.Email, StringComparison.OrdinalIgnoreCase)))
```

Comment:
issue: This causes admin account enumeration. While the UX is not ideal, the way we solve this today is that we send an email informing the target email that they've been invited but can't join since an admin-email can only be connected to one organisation etc.

Reply:
I could be wrong, but `ConsoleAdmins` are shown on the Admins page and having this check wouldn't allow for additional account enumeration.

Reply:
@abergs the only good way to fix it is by allowing the email to be sent. And then do the validation upon clicking the link. Then you avoid enumeration. And the one clicking the link owns the email.

Reply:
Ah, I misunderstood. Thanks!